src/dcc.cpp

```diff
@@ -1,18 +1,66 @@
+#define DUMP_TOKENS 1
+#define DUMP_AST 2
+#define DUMP_IR 3
+#define DUMP_REGISTER_ALLOCATION 4
 
 #include <fstream>
+#include <getopt.h>
 #include <iostream>
 #include <iterator>
 #include <memory>
 #include <string>
 
 #include "compiler.h"
 #include "macros.h"
+#include "option.h"
 #include "parser.h"
 #include "tokenizer.h"
 #include "tokens.h"
 
+std::string Option::outputFile = "out.txt";
+bool Option::dumpTokens = false;
+bool Option::dumpAST = false;
+bool Option::dumpIR = false;
+bool Option::dumpRegisterAllocation = false;
+
 int main(int argc, char** argv) {
-  if (argc != 2) ERR_AND_EXIT("Usage: " << argv[0] << " <file.dlang>");
+  if (argc < 2) ERR_AND_EXIT("Usage: " << argv[0] << "(options) <file.dlang>");
```

Comment:
Move this logic into a helper function that takes `argc` and `argv` and populates `Option::...`?